src/devices/Pn5180/samples/Pn5180sample.cs

```diff
@@ -0,0 +1,423 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Device.Gpio;
+using System.Device.Spi;
+using System.Linq;
+using System.Threading;
+using Iot.Device.Card.CreditCardProcessing;
+using Iot.Device.Card.Mifare;
+using Iot.Device.Ft4222;
+using Iot.Device.Pn5180;
+using Iot.Device.Rfid;
+
+namespace Pn5180sample
+{
+    internal class Program
+    {
+        private static Pn5180 _pn5180;
+
+        public static void Main(string[] args)
+        {
+            Console.WriteLine("Hello Pn5180!");
+            Console.WriteLine($"Choose the device you want to use");
+            Console.WriteLine($"1 for Raspberry Pi");
+            Console.WriteLine($"2 for FT4222");
+            var choice = Console.ReadKey().KeyChar;
+            Console.WriteLine();
+            Console.WriteLine();
+            if (choice == '1')
+            {
+                _pn5180 = Rpi();
+            }
+            else if (choice == '2')
+            {
+                _pn5180 = Ft4222();
+            }
+            else
+            {
+                Console.WriteLine($"Not a correct choice, please choose the device you want to use");
+                return;
+            }
+
+            var (product, firmware, eeprom) = _pn5180.GetVersion();
+            Console.WriteLine($"Product: {product.ToString()}, Firmware: {firmware.ToString()}, EEPROM: {eeprom.ToString()}");
+
+            Console.WriteLine($"Choose what you want to test");
+            Console.WriteLine($"1 dump a full credit card ISO 14443 type B");
+            Console.WriteLine($"2 dump a Mifare IS 14443 type A");
+            Console.WriteLine($"3 EEPROM operations");
+            Console.WriteLine($"4 Radio Frequence operations");
+            Console.WriteLine($"5 Pull ISO 14443 Type A and B cards, display information");
+            Console.WriteLine($"6 Pull ISO 14443 B cards, display information");
+            choice = Console.ReadKey().KeyChar;
+            Console.WriteLine();
+            Console.WriteLine();
+
+            if (choice == '1')
+            {
+                TypeB();
+            }
+            else if (choice == '2')
+            {
+                TypeA();
+            }
+            else if (choice == '3')
+            {
+                Eeprom();
+            }
+            else if (choice == '4')
+            {
+                RfConfiguration();
+            }
+            else if (choice == '5')
+            {
+                PullDifferentCards();
+            }
+            else if (choice == '6')
+            {
+                PullTypeBCards();
+            }
+            else
+            {
+                Console.WriteLine($"Not a valid choice, please choose the test you want to run");
+            }
+        }
+
+        private static Pn5180 Rpi()
```

Comment:
Can you change name to something like HardwareSpi? I suspect there will be new board replacing RPi in 10y and no new dev will even know what Rpi means at that time :)